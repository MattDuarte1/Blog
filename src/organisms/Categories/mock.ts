export default [
  {
    id: '1',
    title: 'javascript',
    image: {
      url: '/javascript',
      alt: 'javascript',
    },
  },
  {
    id: '2',
    title: 'css',
    image: {
      url: '/css',
      alt: 'css',
    },
  },
  {
    id: '3',
    title: 'react',
    image: {
      url: '/react',
      alt: 'react',
    },
  },
  {
    id: '4',
    title: 'tailwind',
    image: {
      url: '/tailwind',
      alt: 'tailwind',
    },
  },
] as const;
