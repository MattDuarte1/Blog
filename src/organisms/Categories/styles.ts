import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  ContainerSize: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        padding: ${theme.spacings.medium};
      }
    `,
  },
  CategoryBodySize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        div {
          width: 100%;
        }
      }
    `,
  },
};

export const Container = styled.section`
  ${({ theme }) => css`
    display: block;
    padding: ${theme.spacings.xxlarge};
    background-color: ${theme.colors.white};
    width: 100%;
    height: 100%;

    ${BreakPoints.ContainerSize['medium'](theme)}
  `}
`;

export const CategoryHeader = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 2rem;
`;

export const CategoryBody = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    min-width: 100%;
    height: 100%;
    gap: 37px;

    ${BreakPoints.CategoryBodySize['small'](theme)}
  `}
`;
