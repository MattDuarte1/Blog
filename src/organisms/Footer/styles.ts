import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    width: 100%;
    padding: 5rem;

    box-shadow: 1px 0 1rem 0 rgba(0, 0, 0, 0.2);

    display: flex;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      gap: 4rem;
      padding: 5rem;
    }
  `}
`;

export const Content = styled.aside`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.small};

    flex: 1;
  `}
`;

export const SocialMediaContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: 2rem;
    justify-content: center;
    width: 40%;
    a {
      color: ${theme.colors.black};
    }

    text-decoration: none;
    color: ${theme.colors.black};
    svg {
      cursor: pointer;
      font-size: 25px;
      transition: all 0.4s ease;

      &:hover {
        transform: scale(1.1);
      }
    }
  `}
`;
