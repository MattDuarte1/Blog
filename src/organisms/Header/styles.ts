import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 86px;
    background-color: ${theme.colors.white};
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.05));

    position: relative;
  `}
`;
