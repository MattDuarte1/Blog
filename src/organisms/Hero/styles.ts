import styled, { css, DefaultTheme, keyframes } from 'styled-components';

const BreakPoints = {
  Container: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteLarge} {
        padding: ${theme.spacings.huge} ${theme.spacings.medium};
      }
    `,
  },
  DescriptionSize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        margin-top: 20px;
        font-size: ${theme.font.sizes.xsmall};
        line-height: normal;
      }
    `,
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        margin-top: 40px;
        font-size: ${theme.font.sizes.small};
        line-height: normal;
      }
    `,
  },
  AsideRight: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        img {
          width: 280px;
          height: 280px;
        }
      }
    `,
  },
};

const cursorEffect = (theme: DefaultTheme) => keyframes`
  0% {
    background-color: ${theme.colors.black};
  }
  100% {
    background-color: ${theme.colors.white};
  }
`;

export const Container = styled.section`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xhuge} ${theme.spacings.medium};
    min-height: 700px;
    display: flex;
    background-color: ${theme.colors.white};
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;

    ${BreakPoints.Container['medium'](theme)}
  `}
`;

export const AsideLeft = styled.aside`
  ${({ theme }) => css`
    max-height: 400px;
    max-width: 664px;
    padding: 25px
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 30px;

    span {
      color: transparent;
      width: 5px;
      height: 100%;
      width: 2px;
      animation: ${cursorEffect(theme)} 1s infinite;
    }
  `}
`;

export const AsideRight = styled(AsideLeft)`
  ${({ theme }) => css`
    justify-content: center;
    display: block;

    ${BreakPoints.AsideRight['medium'](theme)}
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    padding-left: 15px;
    margin-top: 25px;
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 400;
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.darkGrey};
    line-height: 25px;
    margin-left: 20px;
    position: relative;

    &::before {
      content: '';
      position: absolute;
      left: 0px;
      background-color: ${theme.colors.black};
      width: 5px;
      height: 100%;
    }

    ${BreakPoints.DescriptionSize['small'](theme)}
    ${BreakPoints.DescriptionSize['medium'](theme)}
  `}
`;
