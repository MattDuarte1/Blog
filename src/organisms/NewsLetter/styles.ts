import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  Content: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      > h1 {
        font-size: ${theme.font.sizes.medium};
      }

      @media ${theme.media.lteSmall} {
        > h1 {
          font-size: ${theme.font.sizes.small};
        }
        > span {
          font-size: ${theme.font.sizes.xsmall};
        }
      }
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 58px;
    gap: 10px;
    background-color: ${theme.name === 'inverted'
      ? theme.colors.white
      : theme.colors.lightGrey2};
    width: 100%;
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 0px;
    gap: 30px;

    max-width: 696px;

    ${BreakPoints.Content(theme)}
  `}
`;

export const Form = styled.form`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 20px;
  width: 100%;
`;

export const InputForm = styled.input`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
  padding: 21px 21px 21px 25px;

  width: 300px;
  height: 67px;

  /* White */

  background: #fffffe;
  border-radius: 8px;
  border: none;

  &:focus {
    box-shadow: 0 0 0 0;
    outline: 0;
  }
`;
