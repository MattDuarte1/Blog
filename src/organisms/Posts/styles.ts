import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreakPoints = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      padding: 3.5rem;
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 4.4rem;
    background-color: ${theme.colors.white};
    ${MediaBreakPoints.small(theme)}
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: 1.4rem;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(26rem, 2fr));
    gap: ${theme.spacings.large};

    @media (max-width: 1065px) {
      grid-template-columns: repeat(auto-fill, minmax(27rem, 1fr));
    }

    @media (max-width: 705px) {
      justify-content: center;
      grid-template-columns: 0fr;
    }
  `}
`;
