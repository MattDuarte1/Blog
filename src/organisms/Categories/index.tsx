import { forwardRef, useRef } from 'react';
import { Icategory } from '@/interfaces/Category';
import Heading from '@/atoms/Heading';
import CategoryCard from '@/molecules/CategoryCard';
import * as Styled from './styles';
import { useBlogContext } from '@/contexts/Theme/hook';

type CategoriesProps = {
  data?: readonly Icategory[];
};

const Categories = forwardRef<HTMLDivElement, CategoriesProps>(
  ({ data, ...props }, ref) => {
    const {
      changeCategory,
      state: { categorySelected },
    } = useBlogContext();
    const categoryRef = useRef<HTMLDivElement>(null);

    return (
      <Styled.Container id="categoria" ref={ref} {...props}>
        <Styled.CategoryHeader>
          <Heading as="h1" size={'large'} color="black">
            Categorias
          </Heading>
        </Styled.CategoryHeader>
        <Styled.CategoryBody>
          {data?.map((item) => (
            <CategoryCard
              key={item.id}
              ref={categoryRef}
              id={item.id}
              title={item.title}
              image={item.image}
              categorySelected={categorySelected}
              onClick={() => changeCategory(item.title)}
            />
          ))}
        </Styled.CategoryBody>
      </Styled.Container>
    );
  },
);

Categories.displayName = 'Categories Section Component';

export default Categories;
