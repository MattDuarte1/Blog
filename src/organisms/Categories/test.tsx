import Categories from '.';
import { fireEvent, render } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import MockData from './mock';
import { ThemeContext } from '@/contexts/Theme';
import { theme } from '@/styles/theme';
import { ThemeProvider } from 'styled-components';

const changeCategoryMock = jest.fn();

describe('<Categories />', () => {
  it('Should render Categories component ', () => {
    const { container } = renderTheme(<Categories data={MockData} />);

    const categories = container.querySelector('section');
    expect(categories).toBeInTheDocument();
    expect(categories).toHaveAttribute('id', 'categoria');
    expect(categories).toHaveStyle({
      padding: theme.spacings.xxlarge,
      'background-color': theme.colors.white,
      width: '100%',
      height: '100%',
    });
    expect(categories).toHaveStyleRule('padding', theme.spacings.medium, {
      media: theme.media.lteMedium,
    });
  });

  it('Should call changeCategory function when categorycard is clicked', () => {
    const BlogContextValueMock = {
      state: {
        drawerIsOpen: false,
        theme,
        categorySelected: '',
      },
      changeTheme: jest.fn(),
      activeDrawer: jest.fn(),
      changeCategory: changeCategoryMock,
    };
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ThemeContext.Provider value={BlogContextValueMock}>
          <Categories data={MockData} />
        </ThemeContext.Provider>
        ,
      </ThemeProvider>,
    );
    const categoryCard = container.querySelector('article');
    expect(categoryCard).toBeInTheDocument();

    fireEvent.click(categoryCard);
    expect(changeCategoryMock).toHaveBeenCalled();
  });

  it('Should match snapshot categories component', () => {
    const { container } = renderTheme(<Categories data={MockData} />);

    expect(container.firstElementChild).toMatchSnapshot();
  });
});
