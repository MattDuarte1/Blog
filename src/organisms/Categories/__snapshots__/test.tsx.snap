// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> Should match snapshot categories component 1`] = `
.c2 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 3.576rem;
  text-transform: none;
  color: #2B2C34;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 3.576rem;
  width: 220px;
  height: 280px;
  color: #2B2C34;
  background-color: #FFFFFE;
  -webkit-filter: drop-shadow(4px 6px 13px rgba(0,0,0,0.1));
  filter: drop-shadow(4px 6px 13px rgba(0,0,0,0.1));
  border-radius: 8px;
  cursor: pointer;
}

.c4 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: center;
  padding: 0px;
  gap: 30px;
}

.c4:hover {
  background-color: #6246EA;
  color: white;
}

.c0 {
  display: block;
  padding: 6.4rem;
  background-color: #FFFFFE;
  width: 100%;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 2rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 100%;
  height: 100%;
  gap: 37px;
}

@media (max-width:400px) {
  .c2 {
    font-size: 2.4rem;
  }
}

@media (max-width:400px) {
  .c4 {
    width: 80%;
  }
}

@media (max-width:700px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (max-width:400px) {
  .c3 div {
    width: 100%;
  }
}

<section
  class="c0"
  id="categoria"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
      color="black"
    >
      Categorias
    </h1>
  </div>
  <section
    class="c3"
  >
    <article
      aria-details="javascript"
      class="c4"
      id="1"
    >
      <div>
        <img
          alt="javascript"
          src="/javascript"
        />
        <h1>
          javascript
        </h1>
      </div>
    </article>
    <article
      aria-details="css"
      class="c4"
      id="2"
    >
      <div>
        <img
          alt="css"
          src="/css"
        />
        <h1>
          css
        </h1>
      </div>
    </article>
    <article
      aria-details="react"
      class="c4"
      id="3"
    >
      <div>
        <img
          alt="react"
          src="/react"
        />
        <h1>
          react
        </h1>
      </div>
    </article>
    <article
      aria-details="tailwind"
      class="c4"
      id="4"
    >
      <div>
        <img
          alt="tailwind"
          src="/tailwind"
        />
        <h1>
          tailwind
        </h1>
      </div>
    </article>
  </section>
</section>
`;
